With.Fody/ModuleWeaver.cs

```diff
@@ -20,52 +21,57 @@ public ModuleWeaver()
 
     public void Execute()
     {
-        foreach (var type in ModuleDefinition.Types.Where(CanHaveWith))
+        foreach (var type in ModuleDefinition.Types)
         {
-            AddWith(type);
-            RemoveGenericWith(type);
-            LogInfo($"Added method 'With' to type '{type.Name}'.");
+            if (type.GetMethods().Any(m => m.IsPublic && m.Name.StartsWith("With")))
+            {
+                var ctor = GetValidConstructor(type);
+                if (ctor != null)
+                {
+                    AddWith(type, ctor);
+                    RemoveGenericWith(type);
+                    LogInfo($"Added method 'With' to type '{type.Name}'.");
+                }
+            }
         }
     }
 
-    private bool CanHaveWith(TypeDefinition type)
+    private static MethodDefinition GetValidConstructor(TypeDefinition type)
     {
-        if (!type.GetMethods().Any(m => m.IsPublic && m.Name.StartsWith("With")))
-        {
-            return false;
-        }
-
-        var ctors = type.GetConstructors().Where(c => c.IsPublic).ToArray();
-        if (ctors.Length != 1)
-        {
-            return false;
-        }
-
-        var ctor = ctors[0];
-        var parameters = ctor.Parameters;
-        if (parameters.Count < 2)
-        {
-            return false;
-        }
-
-        return parameters.All(par => type.Properties.Any(pro => string.Compare(par.Name, pro.Name, StringComparison.InvariantCultureIgnoreCase) == 0));
+        return type.GetConstructors()
+            .Where(ctor =>
+                ctor.Parameters.Count >= 2 &&
+                ctor.Parameters.All(par =>
+                    type.Properties.Any(pro =>
+                        string.Compare(par.Name, pro.Name, StringComparison.InvariantCultureIgnoreCase) == 0
+                    )
+                )
+            )
+            .FirstOrDefault();
```

Comment:
Well, it least this was my intention. I'm writing more test and I'm finding issues... :(
I can't fix these right now but I hope to be back soon.